src/storage/accessors/InMemoryDataAccessor.ts

```diff
@@ -0,0 +1,170 @@
+import { Readable } from 'stream';
+import arrayifyStream from 'arrayify-stream';
+import { DataFactory } from 'n3';
+import type { NamedNode } from 'rdf-js';
+import { RepresentationMetadata } from '../../ldp/representation/RepresentationMetadata';
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { NotFoundHttpError } from '../../util/errors/NotFoundHttpError';
+import type { MetadataController } from '../../util/MetadataController';
+import { ensureTrailingSlash } from '../../util/Util';
+import type { DataAccessor } from './DataAccessor';
+
+interface DataEntry {
+  data: any[];
+  metadata: RepresentationMetadata;
+}
+interface ContainerEntry {
+  entries: { [name: string]: CacheEntry };
+  metadata: RepresentationMetadata;
+}
+type CacheEntry = DataEntry | ContainerEntry;
+
+class ArrayReadable extends Readable {
+  private readonly data: any[];
+  private idx: number;
+
+  public constructor(data: any[]) {
+    super({ objectMode: true });
+    this.data = data;
+    this.idx = 0;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  public _read(): void {
```

Comment:
```suggestion
  protected _read(): void {
```

Reply:
Has to be public to match the `Readable` interface.